core/src/main/java/org/bitcoinj/core/GetBlocksMessage.java

```diff
@@ -101,7 +101,6 @@ public boolean equals(Object o) {
     @Override
     public int hashCode() {
         int hashCode = (int)version ^ "getblocks".hashCode() ^ stopHash.hashCode();
-        for (Sha256Hash aLocator : locator) hashCode ^= aLocator.hashCode(); // ignores locator ordering
-        return hashCode;
+        return locator.hashCode(hashCode);
```

Comment:
...so don't pass the "intial hashcode" into location.hashCode(...) but instead mix the various hash codes yourself here.  Guava Objects.hashCode() can help.